mesonbuild/cargo/builder.py

```diff
@@ -0,0 +1,283 @@
+    :return: A Token object
+    """
+    return mparser.Token(tid, filename, -1, -1, -1, (-1, -1), value)
+
+
+def string(value: str, filename: str) -> mparser.StringNode:
+    """Build A StringNode
+
+    :param value: the value of the string
+    :param filename: the file that the value came from
+    :return: A StringNode
+    """
+    return mparser.StringNode(_token('string', filename, value))
+
+
+def number(value: int, filename: str) -> mparser.NumberNode:
+    """Build A NumberNode
+
+    :param value: the value of the number
+    :param filename: the file that the value came from
+    :return: A NumberNode
+    """
+    return mparser.NumberNode(_token('number', filename, value))
+
+
+def bool(value: builtins.bool, filename: str) -> mparser.BooleanNode:
+    """Build A BooleanNode
+
+    :param value: the value of the boolean
+    :param filename: the file that the value came from
+    :return: A BooleanNode
+    """
+    return mparser.BooleanNode(_token('bool', filename, value))
+
+
+def array(value: T.List[mparser.BaseNode], filename: str) -> mparser.ArrayNode:
+    """Build an Array Node
+
+    :param value: A list of nodes to insert into the array
+    :param filename: The file the array is from
+    :return: An ArrayNode built from the arguments
+    """
+    args = mparser.ArgumentNode(_token('array', filename, 'unused'))
+    args.arguments = value
+    return mparser.ArrayNode(args, -1, -1, -1, -1)
+
+
+def identifier(value: str, filename: str) -> mparser.IdNode:
+    """Build A IdNode
+
+    :param value: the value of the boolean
+    :param filename: the file that the value came from
+    :return: A BooleanNode
+    """
+    return mparser.IdNode(_token('id', filename, value))
+
+
+def method(name: str, id_: mparser.IdNode,
+           pos: T.Optional[T.List[mparser.BaseNode]] = None,
+           kw: T.Optional[T.Mapping[str, mparser.BaseNode]] = None,
+           ) -> mparser.MethodNode:
+    """Create a method call.
+
+    :param name: the name of the method
+    :param id_: the object to call the method of
+    :param pos: a list of positional arguments, defaults to None
+    :param kw: a dictionary of keyword arguments, defaults to None
+    :return: a method call object
+    """
+    args = mparser.ArgumentNode(_token('array', id_.filename, 'unused'))
+    if pos is not None:
+        args.arguments = pos
+    if kw is not None:
+        args.kwargs = {identifier(k, id_.filename): v for k, v in kw.items()}
+    return mparser.MethodNode(id_.filename, -1, -1, id_, name, args)
+
+
+def function(name: str, filename: str,
+             pos: T.Optional[T.List[mparser.BaseNode]] = None,
+             kw: T.Optional[T.Mapping[str, mparser.BaseNode]] = None,
+             ) -> mparser.FunctionNode:
+    """Create a function call.
+
+    :param name: the name of the function
+    :param pos: a list of positional arguments, defaults to None
+    :param kw: a dictionary of keyword arguments, defaults to None
```

Comment:
Documentation for most params, but filename is missing. :D

This is of course better than the majority of functions throughout the codebase, which have no documentation.